include/cppkafka/message_builder.h

```diff
@@ -421,17 +470,21 @@ class MessageBuilder : public BasicMessageBuilder<Buffer, MessageBuilder> {
     }
 
     template <typename T>
-    void construct_buffer(Buffer& lhs, const T& rhs) {
-        lhs = Buffer(rhs);
+    void construct_buffer(Buffer& lhs, T&& rhs) {
+        lhs = Buffer(std::forward<T>(rhs));
     }
     
+
     MessageBuilder clone() const {
         return std::move(MessageBuilder(topic()).
```

Comment:
I didn't spot this when this was first implemented, but the `std::move` here isn't doing anything and shouldn't be there.

Reply:
Negative on this one. The methods return a `Concrete&` which needs explicit move, otherwise you get the copy-constructor invoked, which doesn't even compile actually because `Concrete&` should be `const Concrete&` to call the copy constructor.

Reply:
Ah, I see. I'd usually prefer constructing an object, calling everything and then just returning it (no need for move there) but that's okay.

Reply:
I will change it as you like.